app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt

```diff
@@ -29,30 +43,77 @@ class EventViewModel() : ViewModel() {
         )
     private val _event = MutableStateFlow<Event>(emptyEvent)
     private val _status = MutableStateFlow<EventStatus>(EventStatus.New)
+    private val allTagsList = MutableStateFlow<List<Tag>>(listOf())
 
-    // placeholder for the list of all tags
-    private val allTagsList = listOf(Tag("1", "tag1"), Tag("2", "tag2"), Tag("3", "tag3"))
+    // initialize async values
+    init {
+        _status.value = EventStatus.Saving // avoid saving before initialization
```

Comment:
Nitpicking: Maybe you could add an `EventStatus.Initializing`, might be more clear...